crates/swc_ecma_minifier/tests/terser/compress/collapse_vars/collapse_rhs_lhs_2/output.js

```diff
@@ -1,6 +1,5 @@
 var b = 1;
-(function f(f) {
-    f = b;
-    f[b] = 0;
+(function f(f1) {
```

Comment:
Don't know why this is renamed -- Could you help @kdy1 ?

Reply:
Because function name (`f`) and the parameter (`f1`) is now in a different scope

Reply:
Previously those were treated as identical to each other

Reply:
As there's `(` in front of function, this is fn expr, not fn decl

Reply:
But why do other colliding ident not got renamed?

Reply:
At where? We reuse variable names in the name mangler, and we only rename declared identifiers in normal var renamer

Reply:
```js
function foo(foo) {}
```
would remain the same after renamer.

Reply:
We are using same analyer for minifier and normal var renamer, and the analyzer knows that you can't refer `foo` (function) from in `foo`

Reply:
Hmm... you are right I'm not sure why is this renamed.

Reply:
I think https://github.com/swc-project/swc/blob/3d393dd709f9a41dc53b15c0fc1edb93997f9d5b/crates/swc_ecma_transforms_base/src/rename/analyzer/mod.rs#L155-L171 is the issue